File: Greedy/week3_1_Split_With_Minimum_Sum/szzng.js
// Runtime 56 ms Beats 73.46%
// Memory 41.3 MB Beats 98.53%

var splitNum = function (num) {
    num = String(num).split('').sort()
    let a = ''
    let b = ''

    for (var i = 0; i < num.length; i += 2) a += num[i]
    for (i = 1; i < num.length; i += 2) b += num[i]

    return Number(a) + Number(b)
};

File: 07_02_Greedy/93_1_Split_With_Minimum_Sum/szzng.py
# Runtime 27 ms Beats 88.73%
# Memory 13.8 MB Beats 94.20%


class Solution:
    def splitNum(self, num: int) -> int:
        num = sorted(list(str(num)))
        a = b = ''

        for i in range(0, len(num), 2): a += num[i]
        for i in range(1, len(num), 2): b += num[i]

        return int(a) + int(b)
